src/test/java/com/example/java/maven/cardGamePeterPan/PlayerTest.java

```diff
@@ -0,0 +1,153 @@
+package com.example.java.maven.cardGamePeterPan;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+
+public class PlayerTest {
+    Player player;
+
+    @BeforeEach
+    void createPlayer() {
+        this.player = new Player("Janusz");
+    }
+
+    @Test
+    void shouldAddCardToPlayersHand() {
+
+//        given
+        Card card = Mockito.mock(Card.class);
+
+//        when
+        player.addCardToHand(card);
+
+//        then
+        assertThat(player.getHandSize()).isEqualTo(1);
+    }
+
+    @Test
+    void shouldTakeCardFromAnotherPlayersHand() {
+
+//        given
+        Player anotherPlayer = new Player("Another");
+        Card card = Mockito.mock(Card.class);
+        anotherPlayer.addCardToHand(card);
+        int cardIndexOutOfBounds = 1;
+
+
+//        when
+
+        player.takeCardFromAnotherPlayersHand(anotherPlayer, 0);
+
+//        then
+
+        assertThat(player.getHandSize()).isEqualTo(1);
+        assertThat(anotherPlayer.getHandSize()).isEqualTo(0);
+        assertThrows(IndexOutOfBoundsException.class, () -> player.takeCardFromAnotherPlayersHand(anotherPlayer, cardIndexOutOfBounds));
+
+
+    }
+
+    @Test
+    void shouldReturnTrueIfDiscardingBySuitIsPossible() {
+
+//        given
+        Card card1 = new Card(Card.Suit.CLUB, Card.Rank.ACE);
+        Card card2 = new Card(Card.Suit.SPADE, Card.Rank.JACK);
+        Card card3 = new Card(Card.Suit.CLUB, Card.Rank.TEN);
+
+//        when
+        player.addCardToHand(card1);
+        player.addCardToHand(card2);
+        player.addCardToHand(card3);
```

Comment:
MINOR:
adding cards is a part of test preparation - `given` section. In this test you are testing `isDiscardingCardsPossible()` method

Reply:
Done